packages/labs/compiler/src/lib/ast-fragments.ts

```diff
@@ -111,16 +252,43 @@ export const createCompiledTemplateResult = ({
 export const createTemplateParts = ({
   f,
   parts,
+  partIdentifiers,
 }: {
   f: ts.NodeFactory;
   parts: TemplatePart[];
+  partIdentifiers: AttributePartConstructorAliases;
 }) =>
   f.createArrayLiteralExpression(
     parts.map((part) => {
       const partProperties = [
         f.createPropertyAssignment('type', f.createNumericLiteral(part.type)),
         f.createPropertyAssignment('index', f.createNumericLiteral(part.index)),
       ];
+      if (
+        part.type === PartType.ATTRIBUTE &&
+        (part.ctorType === PartType.ATTRIBUTE ||
```

Comment:
isn't this second clause always true? what other `ctorType` could there be?

Reply:
... Yes it is. I'll be removing it. Great catch!